official/transformer/transformer_main.py

```diff
@@ -269,18 +328,26 @@ def train_schedule(
       train_eval_iterations = INF
 
   # Loop training/evaluation/bleu cycles
-  for i in xrange(train_eval_iterations):
+  for i in xrange(schedule_manager.train_eval_iterations):
     tf.logging.info("Starting iteration %d" % (i + 1))
 
     # Train the model for single_iteration_train_steps or until the input fn
     # runs out of examples (if single_iteration_train_steps is None).
     estimator.train(
-        dataset.train_input_fn, steps=single_iteration_train_steps,
+        dataset.train_input_fn,
+        steps=schedule_manager.single_iteration_train_steps,
         hooks=train_hooks)
 
-    eval_results = estimator.evaluate(dataset.eval_input_fn)
+    schedule_manager.post_train()
```

Comment:
This is mysterious. I would prefer this belong to a TPU util that handle this operation, rather than a generic magic-box.

Reply:
Happily this is all moot, as infra has fixed the underlying issue. Yay!